### PR TITLE
Reorganized the lambda code to have only one call, possibly throwing an exception

### LINKED ISSUE #159: Sonar. Refactor the code of the lambda to have only one invocation possibly throwing a runtime exception
Need to fix code smells https://sonarcloud.io/project/issues?resolved=false&severities=MAJOR&id=mfvanek_pg-index-health

### PR DESCRIPTION
**Pull request by EnKarin**
Closes #159

### CODE REVIEW

File: src/test/java/io/github/mfvanek/pg/model/index/ForeignKeyTest.java [hunk 1]
```diff
@@ -98,7 +98,8 @@ void withInvalidArguments() {
                 .isInstanceOf(NullPointerException.class);
         assertThatThrownBy(() -> ForeignKey.of("t", "c_t_order_id", null))
                 .isInstanceOf(NullPointerException.class);
-        assertThatThrownBy(() -> ForeignKey.of("t", "c_t_order_id", Collections.emptyList()))
+        final List<Column> emptyList = Collections.emptyList();
```
Comment (mfvanek) [inline src/test/java/io/github/mfvanek/pg/model/index/ForeignKeyTest.java:101]: Please rename `emptyList` to `columns`

File: src/test/java/io/github/mfvanek/pg/model/index/DuplicatedIndexesTest.java [hunk 2]
```diff
@@ -101,19 +101,24 @@ void withoutIndexes() {
         assertThatThrownBy(() -> DuplicatedIndexes.of(null))
                 .isInstanceOf(NullPointerException.class)
                 .hasMessage("duplicatedIndexes cannot be null");
-        assertThatThrownBy(() -> DuplicatedIndexes.of(Collections.emptyList()))
+
+        final List<IndexWithSize> emptyList = Collections.emptyList();
```
Comment (mfvanek) [inline src/test/java/io/github/mfvanek/pg/model/index/DuplicatedIndexesTest.java:105]: Please rename emptyList to indexes

File: src/test/java/io/github/mfvanek/pg/model/index/DuplicatedIndexesTest.java [hunk 3]
```diff
@@ -101,19 +101,24 @@ void withoutIndexes() {
...
+        assertThatThrownBy(() -> DuplicatedIndexes.of(emptyList))
                 .isInstanceOf(IllegalArgumentException.class)
                 .hasMessage("duplicatedIndexes cannot be empty");
-        assertThatThrownBy(() -> DuplicatedIndexes.of(Collections.singletonList(IndexWithSize.of("t", "i", 1L))))
+
+        final List<IndexWithSize> singletonList = Collections.singletonList(IndexWithSize.of("t", "i", 1L));
```
Comment (mfvanek) [inline src/test/java/io/github/mfvanek/pg/model/index/DuplicatedIndexesTest.java:110]: Please rename singletonList

File: src/test/java/io/github/mfvanek/pg/model/index/DuplicatedIndexesTest.java [hunk 4]
```diff
@@ -198,14 +203,17 @@ void newFactoryConstructor() {
...
-        assertThatThrownBy(() -> DuplicatedIndexes.of(IndexWithSize.of("t", "i1", 1L), null))
+
+        final IndexWithSize indexWithSize = IndexWithSize.of("t", "i1", 1L);
+        assertThatThrownBy(() -> DuplicatedIndexes.of(indexWithSize, null))
                 .isInstanceOf(NullPointerException.class)
                 .hasMessage("secondIndex cannot be null");
+
+        final IndexWithSize firstIndex = IndexWithSize.of("t", "i1", 1L);
+        final IndexWithSize secondIndex = IndexWithSize.of("t", "i2", 2L);
+        final IndexWithSize[] otherIndexes = {null, IndexWithSize.of("t", "i4", 4L)};
```
Comment (mfvanek) [inline src/test/java/io/github/mfvanek/pg/model/index/DuplicatedIndexesTest.java:214]: Why not `final IndexWithSize fourthIndex = IndexWithSize.of("t", "i4", 4L)` ?

**Review by mfvanek — CHANGES_REQUESTED** [review]

File: src/test/java/io/github/mfvanek/pg/model/index/ForeignKeyTest.java (hunk 1 above)
Comment (EnKarin) [reply src/test/java/io/github/mfvanek/pg/model/index/ForeignKeyTest.java]: Renamed

File: src/test/java/io/github/mfvanek/pg/model/index/DuplicatedIndexesTest.java (hunk 2 above)
Comment (EnKarin) [reply src/test/java/io/github/mfvanek/pg/model/index/DuplicatedIndexesTest.java]: Renamed

File: src/test/java/io/github/mfvanek/pg/model/index/DuplicatedIndexesTest.java (hunk 3 above)
Comment (EnKarin) [reply src/test/java/io/github/mfvanek/pg/model/index/DuplicatedIndexesTest.java]: Renamed

File: src/test/java/io/github/mfvanek/pg/model/index/DuplicatedIndexesTest.java (hunk 4 above)
Comment (EnKarin) [reply src/test/java/io/github/mfvanek/pg/model/index/DuplicatedIndexesTest.java]: Array was deleted and two separate variables were created

**Pull request merged**